text/086-reduce-barriers-to-contributing-to-stdlib.md

```diff
@@ -0,0 +1,250 @@
+# Compat Strategy
+
+We should make every effort to escape the rigidity trap the stdlib is currently caught in,
+while maintaining the status of stdlib as a reliable piece of infrastructure to depend upon and followed through its development.
+Stability in the sense of avoiding change is not a goal.
+The median change to stdlib will require backwards-compatible changes in multiple CI developments.
+These changes would be routinely merged by maintainers of these developments,
+and co-developed by them if necessary due to complexity of their code, unusual requirements, or otherwise.
+Once-and-over-with strategies for CI-wide changes should be preferred of phased rollouts whenever practical,
+to avoid needless code churn and developer toil in both stdlib and downstream.
+In cases where an atomic replacement is infeasible, but there are strong reasons for making the change,
+a pilot rollout in representative developments should precede an add-deprecate-replace-remove cycle.
+No step in this process should involve waiting for another Coq release, or any specific amount of time --
+if we have ported the CI and are confident that other developments can be ported, the change should proceed without delay.
+
+We should also invest in tooling to scale this process, automating inlining of deprecated definitions, fine-grained dependency tracking for local CI builds and automation for submitting and following up on downstream pull requests.
+Specific build-tooling improvements are outside the scope of this CEP, it is acknowledged they are comparably important to process improvements.
+
+## Draft Additions to [Contributing Guide](https://github.com/coq/coq/blob/master/CONTRIBUTING.md#additional-notes-for-pull-request-reviewers-and-assignees)
+
+Every change to Coq, including the standard library, should be tested against the CI developments to ensure that it works for them.
+If CI developments need changes two work with the updated Coq,
+the author of the PR should work with the maintainers with these developments to incorporate the changes.
+Coq changes with complex compat requirements should come with a porting plan as a part of the design,
+and that porting plan should be validated and refined by porting the CI developments.
+
+If a satisfactory porting plan is found and the CI is ported, no deprecation delay is be required, even for changes that require porting.
+Non-CI developments are expected to follow the porting plan developed as a part of the PR and validated on the CI.
+On the other hand, changes for which porting requires an extended period of time, or needs multiple phases for a technical reason,
+should start by addition of new functionality and deprecation of the old one, ideally with a target date/release for removal.
+Any change that is known or expected to break any development in a hard-to-fix way should be discussed with the Coq team and the maintainers of that development (perhaps as a CEP), but with the Coq team making the final decision.
```

Comment:
On the other hand, this paragraph (which is confirming my interpretation above) is also kind of making explicit a practice that already exists: what matters is how much effort of porting will be needed, and for simple things, skipping a deprecation may be fine (but it should be carefully evaluated).